components/script/dom/htmldialogelement.rs

```diff
@@ -124,4 +125,25 @@ impl HTMLDialogElementMethods<crate::DomTypeHolder> for HTMLDialogElement {
             .dom_manipulation_task_source()
             .queue_simple_event(target, atom!("close"));
     }
+
+    // https://html.spec.whatwg.org/multipage/interactive-elements.html#dom-dialog-requestclose
+    fn RequestClose(&self, return_value: Option<DOMString>) {
+        let element = self.upcast::<Element>();
+        let target = self.upcast::<EventTarget>();
+
+        if !element
+            .has_attribute(&local_name!("open"))
+        {
+            return;
+        }
+
+        let window = self.owner_window();
+        let event = Event::new(window.upcast(), atom!("cancel"), EventBubbles::DoesNotBubble, EventCancelable::Cancelable, CanGc::note());
+        let event_status = event.fire(target, CanGc::note());
```

Comment:
if you add a `canGc` entry for this method in `Bindings.conf` you'll get a `CanGc` argument that you can use here